### PR TITLE
Add Sentry CSP reporting support

### PR DESCRIPTION
**Pull request by ChristophWurst**
- [x] Requires #58 
- [x] Doc updates

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Still some stuff to do: https://github.com/ChristophWurst/nextcloud_sentry/pull/60#issue-237638484, but feel free to complete this :wink:

**Comment by ChristophWurst** [comment]
Tested and works on my dev env with FF.

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -45,15 +47,20 @@ public function __construct($urlParams = []) {
...
 		}
 		$publicDsn = $config->getSystemValueString('sentry.public-dsn', '');
 		$this->setInitialState($publicDsn);
-		if ($publicDsn !== '') {
-			$this->addCsp($publicDsn);
-		}
+
+		/** @var IEventDispatcher $dispatcher */
+		$dispatcher = $container->query(IEventDispatcher::class);
+		$dispatcher->addListener(AddContentSecurityPolicyEvent::class, function (AddContentSecurityPolicyEvent $event) use ($reportUrl, $publicDsn) {
```
Comment (rullzer) [inline lib/AppInfo/Application.php]: nice :wink:

**Comment by ChristophWurst** [comment]
Bildschirmfoto von 2019-08-26 21-17-40

an example report

**Pull request merged**